saas/models.py

```diff
@@ -1314,7 +1315,8 @@ def create_order(self, nb_periods, prorated_amount=0,
             ends_at = self.plan.end_of_period(self.ends_at, nb_periods)
             descr = describe_buy_periods(self.plan, ends_at, nb_periods)
             if discount_percent:
-                descr += ' - a %d%% discount' % discount_percent
+                descr += ' - a %d%% discount (code: %s)' % (
+                    discount_percent, discount_descr)
```

Comment:
@smirolo you are correct about `discount_descr`, but that would only happen if `discount_descr` was `None` or not provided, and `discount_percent` was not `None`. This is what I was referring to about trying to copy the existing style - I figured that because `discount_percent` was passed, instead of say, a `Coupon` object, then it was desirable to stick with separate args. I also considered making a single tuple arg, `discount`, which would be `(percent, description)`.

I think that, from a MVC purist point of view, the model code shouldn't make assumptions about the presentation layer, and this description string qualifies as presentation. What if it needs to be localized? How does the wrapper site customize it, say, if it wants to show " - you saved $103.00 (50%) by using code DIS50" instead?

This agrees with moving generation of these strings out to the view layer, but maybe suggests that there should be a templating system for generating them in the first place, driven by configuration? And then the billing view would ask the `Subscription` model for its options as a method, and pass the cart options and the currently-applied coupon (if any) to the line item renderer, which would produce the strings for display.

Overkill?